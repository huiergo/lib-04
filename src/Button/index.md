---
nav:
  title: Components
  path: /components
group:
  title: components
---

## Button

Demo:

```tsx
import React from 'react';
import { Button } from 'lib-04';
import GlobalConfig from '../GlobalConfig';

export default () => (
  <GlobalConfig globalColor="red">
    <Button className="self-btn" type="primary" style={{ fontSize: 16 }}>
      按钮1
    </Button>
  </GlobalConfig>
);
```

More skills for writing demo: https://d.umijs.org/guide/basic#write-component-demo
