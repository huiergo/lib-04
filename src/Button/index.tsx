import React, { useContext } from 'react';
import { ButtonProps } from './interface';
import classNames from 'classnames';
import './index.less';
import { globalCtx } from '../GlobalConfig';
import { GlobalConfigProps } from '../GlobalConfig/interface';

function Button(props: ButtonProps) {
  const { className, type, disabled, circle, dashed, loading, icon, handleClick, ...other } = props;
  const { globalColor } = useContext(globalCtx) as GlobalConfigProps;

  console.log('globalColor---', globalColor);
  // todo
  const prefixCls = 'lib-04';
  const darkTheme = true;
  // classNames 直接传值（可以是常量，也可以是变量），& 传递对象（代表有无这个class）
  const classNamesss = classNames(
    'bfd-btn',
    prefixCls,
    className,
    { 'lib-button-dark': darkTheme },
    // [`${prefixCls}--${theme}`]: !!theme,
  );
  const btnClick: React.MouseEventHandler<HTMLDivElement> = (event) => {
    handleClick && handleClick(event);
  };
  return (
    <div className={classNamesss} {...other} onClick={btnClick}>
      Button
      {/* <div style={{background:globalColor}}> 我是 globalColor</div> */}
    </div>
  );
}

export default Button;
