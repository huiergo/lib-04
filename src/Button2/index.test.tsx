import React from 'react';
// import { render, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Button from './index';
import { render } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('Button', () => {
  it('renders correctly', () => {
    const wrapper = render(<Button>foo</Button>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  // it('icon', () => {
  //   const wrapper = render(
  //     <Button icon={<img alt="" src="https://zarm.design/images/logo.ce68565d.svg" />}>foo</Button>,
  //   );
  //   expect(toJson(wrapper)).toMatchSnapshot();
  // });

  // it('disabled', () => {
  //   const wrapper = render(<Button disabled>foo</Button>);
  //   expect(toJson(wrapper)).toMatchSnapshot();
  // });

  // it('onClick', () => {
  //   // const onClick = jest.fn();
  //   const onclick=()=>console.log('diaoyognle')
  //   const wrapper = shallow(<Button onClick={onclick}>foo</Button>);
  //   wrapper.simulate('click');
  //   expect(onclick).toBeCalled();
  // });

  // it('onClick when disabled', () => {
  //   const onClick = jest.fn();
  //   const wrapper = shallow(
  //     <Button disabled onClick={onClick}>
  //       foo
  //     </Button>,
  //   );
  //   wrapper.simulate('click');
  //   expect(onClick).not.toBeCalled();
  // });

  // it('onTouchStart in NativeButton', () => {
  //   const wrapper = shallow(<Button>foo</Button>);
  //   wrapper.simulate('touchstart');
  // });
});
