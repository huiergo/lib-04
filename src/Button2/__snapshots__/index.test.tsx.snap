// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button disabled 1`] = `
<div
  class="btn test btn_disabled"
>
  foo
</div>
`;

exports[`Button renders correctly 1`] = `
<div
  class="btn"
>
  foo
</div>
`;
